### ISSUE TITLE
flex-grow not filling up entire parent

### ISSUE THREAD

**Issue by 8176135** [fschutt/azul#165]
## Description

flex-grow doesn't seem to want to fill up the entire parent dom, example:

Layout:
```
Dom::div().with_id("first_line")
	.with_child(Label::new("Child IP: ").dom().with_id("child_ip_label"))
	.with_child(Dom::div().with_id("other"))
```
CSS:
```
#first_line {
    margin: 0px;
    padding: 0px;
    background-color: green;
}

#other {
    flex-grow: 3;
    background-color: red;
}

#child_ip_label {
    flex-grow: 1;
}
```
We get this:
![2019-04-29 21_38_23-Azul App](https://user-images.githubusercontent.com/11830488/56888168-7aa23e00-6ac7-11e9-8e30-a41672f3a4d1.png)

The issue is even worse when you add `flex-direction: row` to `#first_line`:

![2019-04-29 21_42_18-Azul App](https://user-images.githubusercontent.com/11830488/56888284-b3421780-6ac7-11e9-9f71-80c51ee28cf2.png)

What's even weirder is that when flex direction is column, the gap at the bottom seems to have a constant height when rescaling the window. However, when flex-direction is row, the gap at the right side partially scales with the window size.

**Edit:**  After further testing, it seems we *don't* need 2 child elements for the `flex-direction: column` case, just one will exhibit the same behavior: i.e. a constant height gap between the bottom of the parent and the child.

**HOWEVER** with just one child element, it fills up the parent just fine when the parent uses `flex-direction: row`

(One child as in: 
```
Dom::div().with_id("first_line")
	.with_child(Label::new("Child IP: ").dom().with_id("child_ip_label"))
```

```
#first_line {
    margin: 0px;
    padding: 0px;
    background-color: green;
}
#child_ip_label {
    background-color: red;
}
```
)
## Version / OS

* azul version:
Latest commit on master as of this post: (69e4912)

* Operating system:
Windows 10

## Steps to Reproduce

As described above, everything else is the same as the hello_world example, with the exception of the css override, i.e.: 
```
let window = app.create_window(WindowCreateOptions::default(), css::override_native( include_str!("main.css")).unwrap()).unwrap();
```

**Comment by fschutt** [comment]
This was originally reported already in #155, but this issue has the better description of the bug. As for the bug, it has to do with the text itself, the "Child IP" text. In the near future, I want to move the whole layout system towards using [stretch](https://github.com/vislyhq/stretch) - which is way better tested than this crate and has a better implentation of the actual flexbox algorithm. When I started azul, stretch wasn't available, so I had to write my own layout solver, but that works only for simpler cases. This bug does have a large priority, because it's also messing up my apps, it should be fixed soon.

**Comment by JAicewizard** [comment]
thanks for providing more information than I did, I just noticed it and wanted to report it but did't(/don't) have enough (rust) experience to do any testing.

**Comment by fschutt** [comment]
I could reproduce this bug, but right now it has more to do with the CSS not being applied correctly rather than with the layout.

The complete code to reproduce this bug on the current master (https://github.com/fschutt/azul/commit/21d209b10bf2b24987165ca8552ff622e50e5177) is:

```rust
#![windows_subsystem = "windows"]

use azul::{
    app::{App, AppConfig},
    window::WindowCreateOptions,
    style::StyledDom,
    css::Css,
    dom::Dom,
    callbacks::{RefAny, LayoutCallbackInfo},
};
use azul_widgets::label::Label;

#[derive(Debug)]
struct Data { }

extern "C" fn layout(_: &mut RefAny, _info: LayoutCallbackInfo) -> StyledDom {
    let mut styled_dom = Dom::div()
    .style(Css::from_string("div {
        margin: 0px;
        padding: 0px;
        background: green;
        flex-grow: 1;
    }".into()));

    let child1 = Label::new("Child IP: ").dom();
    // child1.restyle(Css::from_string(...));

    let child2 = Dom::div()
    .style(Css::from_string("div {
        flex-grow: 3;
        background: red;
    }".into()));

    styled_dom.append_child(child1);
    styled_dom.append_child(child2);

    styled_dom
}

fn main() {
    let data = RefAny::new(Data { });
    let app = App::new(data, AppConfig::default());
    app.run(WindowCreateOptions::new(layout));
}
```

Result:

![image](https://user-images.githubusercontent.com/12084016/114427154-1648ee00-9bbb-11eb-9725-bccc6dc5064a.png)

**Comment by fschutt** [comment]
This issue has been fixed now (1.0.0-alpha1)
